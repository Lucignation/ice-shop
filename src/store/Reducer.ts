import { createSlice } from '@reduxjs/toolkit';
import type { PayloadAction } from '@reduxjs/toolkit';
import { IUser, IAccount } from '../common/interfaces';
import { IProduct } from '../common/product.interface';
import { ICart } from '../common/interfaces';

const initialState: IAccount = {
  user: {
    id: 0,
    email: '',
    username: '',
    password: '',
    name: {
      firstname: '',
      lastname: '',
    },
    address: {
      city: '',
      street: '',
      number: 0,
      zipcode: '',
      geolocation: {
        lat: '',
        long: '',
      },
    },
    phone: '',
  },
  token: '',
  products: [],
  product: {
    id: 0,
    title: '',
    price: '',
    category: '',
    description: '',
    image: '',
  },
  cart: [],
};

export const userSlice = createSlice({
  name: 'user',
  initialState,
  reducers: {
    setUser: (state: IAccount, action: PayloadAction<IAccount>) => {
      //   localStorage.setItem('token', action.payload.token);
      //   // localStorage.removeItem('token');
      //   state.user = action.payload.msg;
    },
    setLogin: (state: IAccount, action: PayloadAction<IAccount>) => {
      localStorage.setItem('token', action.payload.token);
      state.token = action.payload.token;
    },
    GetProducts: (state: IAccount, action: PayloadAction<IProduct[]>) => {
      return {
        ...state,
        products: action.payload,
      };
    },
    RemoveItem: (state: IAccount, action: PayloadAction<IProduct>) => {
      return {
        ...state,
        cart: state.cart.filter((product) => product.id !== action.payload.id),
      };
    },
    AddItem: (state: IAccount, action: PayloadAction<IProduct>) => {
      return {
        ...state,
        cart: state.cart.concat(action.payload),
      };
    },

    AddUser: (state: IAccount, action: PayloadAction<IUser>) => {
      return {
        ...state,
        user: action.payload,
      };
    },

    UpdateUser: (state: IAccount, action: PayloadAction<IUser>) => {
      return {
        ...state,
        user: action.payload,
      };
    },
    logOut: () => {
      localStorage.removeItem('token');
    },
  },
});

// Action creators are generated for each case reducer function
export const {
  setUser,
  logOut,
  setLogin,
  GetProducts,
  RemoveItem,
  AddItem,
  AddUser,
  UpdateUser,
} = userSlice.actions;

export default userSlice.reducer;
