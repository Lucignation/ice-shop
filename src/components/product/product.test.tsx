import Product from './product.component';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { store } from '../../store/store';
const mockProps = {
  id: 1,
  title: 'product title',
  price: '10',
  category: 'product category',
  image: 'product image url',
  description: 'product description',
};

describe('Product: ', () => {
  it('renders a snapshot of <Product /> when each item is passed to the component', () => {
    const product = renderer
      .create(
        <Provider store={store}>
          <Product product={mockProps} />
        </Provider>
      )
      .toJSON();
    expect(product).toMatchSnapshot();
  });
});
