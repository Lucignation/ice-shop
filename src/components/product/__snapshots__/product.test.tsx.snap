// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product:  renders a snapshot of <Product /> when each item is passed to the component 1`] = `
<div
  className="product_container"
>
  <div
    className="product_image"
  >
    <img
      alt="product title"
      src="product image url"
    />
  </div>
  <div
    className="product_info_container"
  >
    <h4
      className="product_title"
    >
      product title...
    </h4>
    <div
      className="product_mid_main"
    >
      <span
        className="product_price"
      >
        $
        10
      </span>
      <span
        className="product_category"
      >
        product category
      </span>
    </div>
    <p
      className="product_description"
    >
      product description...
    </p>
  </div>
  <p
    className="btn btn-outline-secondary add_to_cart_btn"
    onClick={[Function]}
  >
    Add to cart
  </p>
</div>
`;
